contracts/Goellars.sol

```diff
@@ -69,10 +69,6 @@ contract Goellars is IERC20, MinterRole {
       _pullAndMint(from, msg.sender, amountToMint);
     }
     if (amountToMint < value) {
```

Comment:
this condition is always true (unless value is 0) and thus can be removed

Reply:
wrong.. it will be `false` if `_balances[from]` is 0. So if no balance on from, no transfer happens — probably a bug

Reply:
the idea is to transferFrom what is there, and if it is not enough, mint the rest.

Reply:
I understand this. But still the condition is useless/broken

Reply:
> I understand this. But still the condition is useless/broken

the condition says: if the guys had more than nothing before minting, then transfer that something now.

Reply:
but `amountToMint` will never be bigger than `value` here by definition (see line 64), why check this then?

Reply:
ok, this makes sense

> the condition is there so that we don't try to transfer 0

though pretty obscure. I would replace with `require(value > 0, "Value cannot be 0")` as the very first line of the function

Reply:
besides, in case of 0 value the function will fail on line 64 anyway 🤷‍♂️